Sources/TokamakLint/Rules/RenderGetsHooksRule.swift

```diff
@@ -15,38 +15,49 @@ struct RenderGetsHooksRule: Rule {
   )
 
   public static func validate(visitor: TokenVisitor) -> [StyleViolation] {
+    var violations: [StyleViolation] = []
+
     do {
-      let renderFunction = try visitor.root.getOneRender(at: visitor.path)
-      guard let codeBlock = renderFunction.firstParent(
-        of: SyntaxKind.codeBlockItem.rawValue
-      ) else { return [StyleViolation(
-        ruleDescription: OneRenderFunctionRule.description,
-        location: Location(
-          file: visitor.path,
-          line: renderFunction.range.startRow,
-          character: renderFunction.range.startColumn
-        )
-      )] }
-      guard let functionSignature = codeBlock.firstChild(
-        of: SyntaxKind.functionSignature.rawValue
-      ) else { return [] }
+      // search for render function
+      let structs = visitor.root.components(hookedComponentProtocols)
+      guard !structs.isEmpty else { return [] }
```

Comment:
this check is redundant as the following `forEach` won't execute anything anyway if `structs` is empty